### PR TITLE
fix: INTMDB-818 cluster private endpoint conn string

### PR DESCRIPTION
**Pull request by Adelmar92**
## Proposed changes

This pull request introduces modifications to the cluster resource connection string properties to address an issue where users were unable to retrieve the private endpoint connection strings. The previous implementation outputted the private endpoint connection strings within a list of objects, making it difficult for users to access the necessary information. To improve usability and clarity, the following changes have been made:

1. Removed the PrivateEndpoint property.
2. Added two new properties: PrivateEndpoints and PrivateEndpointsSrv
     - Both properties are lists of strings.
     - These lists contain the connection strings for the private endpoints.

JiraTicket : [INTMDB-818](https://jira.mongodb.org/browse/INTMDB-818)

Testing:

stack testing
i created a stack with this outputs: 
image

image

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [x] cfn invoke for each of CRUDL/cfn test
- [x] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [x] Published to AWS private registry
- [x] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [x] Deleted stack to ensure resources are deleted
- [x] Created multiple resources in same stack
- [x] Validated in Atlas UI
- [x] Included screenshots

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

**Review by maastha — COMMENTED** [review]
Hi @Adelmar92 could you please add ticket link to the PR description? otherwise LGTM!

File: cfn-resources/cluster/cmd/resource/resource.go [hunk 1]
```diff
@@ -676,32 +676,27 @@ func flattenBiConnectorConfig(biConnector *mongodbatlas.BiConnector) *BiConnecto
...
-		prvEndpoints = append(prvEndpoints, pe)
+func flattenPrivateEndpoint(pes []mongodbatlas.PrivateEndpoint) (privateEndpoints, privateEndpointsServ []string) {
+	privateEndpoints = make([]string, len(pes))
+	privateEndpointsServ = make([]string, len(pes))
+	for i := range pes {
+		privateEndpoints[i] = pes[i].ConnectionString
+		privateEndpointsServ[i] = pes[i].SRVConnectionString
 	}
-	return prvEndpoints
+	return privateEndpoints, privateEndpointsServ
```
Comment (andreaangiolillo) [inline cfn-resources/cluster/cmd/resource/resource.go:699]:
```suggestion
	return privateEndpoints, privateEndpointsSrv
```

File: cfn-resources/cluster/cmd/resource/resource.go [hunk 2]
```diff
@@ -676,32 +676,27 @@ func flattenBiConnectorConfig(biConnector *mongodbatlas.BiConnector) *BiConnecto
...
-		pe := PrivateEndpoint{
-			ConnectionString:    &pes[ind].ConnectionString,
-			SRVConnectionString: &pes[ind].SRVConnectionString,
-			Type:                &pes[ind].Type,
-			Endpoints:           flattenEndpoints(pes[ind].Endpoints),
-		}
-		prvEndpoints = append(prvEndpoints, pe)
+func flattenPrivateEndpoint(pes []mongodbatlas.PrivateEndpoint) (privateEndpoints, privateEndpointsServ []string) {
+	privateEndpoints = make([]string, len(pes))
+	privateEndpointsServ = make([]string, len(pes))
```
Comment (andreaangiolillo) [inline cfn-resources/cluster/cmd/resource/resource.go:694]:
```suggestion
	privateEndpointsSrv = make([]string, len(pes))
```

File: cfn-resources/cluster/cmd/resource/resource.go [hunk 3]
```diff
@@ -676,32 +676,27 @@ func flattenBiConnectorConfig(biConnector *mongodbatlas.BiConnector) *BiConnecto
 
 func flattenConnectionStrings(clusterConnStrings *mongodbatlas.ConnectionStrings) (connStrings *ConnectionStrings) {
 	if clusterConnStrings != nil {
+		privateEndpoints, privateEndpointsServ := flattenPrivateEndpoint(clusterConnStrings.PrivateEndpoint)
```
Comment (andreaangiolillo) [inline cfn-resources/cluster/cmd/resource/resource.go:679]:
```suggestion
		privateEndpoints, privateEndpointsSrv := flattenPrivateEndpoint(clusterConnStrings.PrivateEndpoint)
```

**Review by andreaangiolillo — APPROVED** [review]

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**